version/string.go

```diff
@@ -4,37 +4,54 @@
 package version
 
 import (
+	"encoding/json"
 	"fmt"
 	"runtime"
+	"strconv"
 	"strings"
 )
 
-var (
-	// String is displayed when CLI arg --version is used
-	String string
+// GitCommit is set in the build script at compile time
+var GitCommit string
 
-	// GitCommit is set in the build script at compile time
-	GitCommit string
-)
+// Versions contains the versions relevant to a build of avalanchego. In
+// addition to supporting construction of the strings displayed by
+// --version and --json-version, it can be used to unmarshal the output
+// of --json-version.
+type Versions struct {
```

Comment:
I don't really love this name... Would `version.Description` be better than `version.Versions`? I don't really love that either tbh.